src/shape/point.js

```diff
@@ -2,32 +2,121 @@
  * Copyright (c) 2017 NAVER Corp.
  * billboard.js project is licensed under the MIT license
  */
+import {select as d3Select} from "d3";
 import ChartInternal from "../internals/ChartInternal";
-import {isFunction, isObjectType, extend} from "../internals/util";
+import {isFunction, isObjectType, extend, notEmpty} from "../internals/util";
 
 extend(ChartInternal.prototype, {
 	hasValidPointType(type) {
 		return /^(circle|rect(angle)?|polygon|ellipse)$/i.test(type || this.config.point_type);
 	},
 
-	hasValidPointDrawMethods() {
-		const pointType = this.config.point_type;
+	hasValidPointDrawMethods(type) {
+		const pointType = type || this.config.point_type;
 
 		return isObjectType(pointType) &&
 			isFunction(pointType.create) && isFunction(pointType.update);
 	},
 
+	insertPointInfoDefs(point, id) {
+		const $$ = this;
+		const defs = $$.svg.select("defs");
+
+		if (defs.size < 1) {
+			return;
+		}
+
+		const html = defs.html();
+
+		// Add the point node into <defs>
+		defs.html(`${html}${point}`);
```

Comment:
IMO, touching whole defs isn't really good idea. There will be more efficient approach on this? Just appending the necessary node?

Reply:
@netil you're right.
I've updated this function to use `defs.node().appendChild` instead.
Tell me what you think